src/Build.UnitTests/Evaluation/ItemEvaluation_Tests.cs

```diff
@@ -507,5 +507,17 @@ public void LongIncludeChain()
 
             Assert.Equal(INCLUDE_COUNT, items.Count);
         }
+
+        // see https://github.com/Microsoft/msbuild/issues/2069
+        [Fact]
+        public void ImmutableListBuilderBug()
+        {
+            var content = @"<i Include=""0;x1;x2;x3;x4;x5;6;7;8;9""/>
+                            <i Remove=""x*""/>";
+
+            IList<ProjectItem> items = ObjectModelHelpers.GetItemsFromFragment(content);
+
+            Assert.Equal("0;6;7;8;9", String.Join(";", items.Select(i => i.EvaluatedInclude)));
```

Comment:
There's an `Assert.Equal<T>(IEnumerable<T> expected, IEnumerable<T> actual, IEqualityComparer<T> comparer)` that gives nice error messages for this situation.